### DISCUSSION TITLE
"Feature Not Implemented or Requested valid for version." Error in ZP_AISDKDEMO_CHATCOMPL_SIMPLE

### DISCUSSION THREAD

**Discussion by Dev4Develop** [microsoft/aisdkforsapabap#19] [Ideas]
I was trying MS AI SAP SDK and ran into this error when ZP_AISDKDEMO_CHATCOMPL_SIMPLE is executed. Pls suggest to rectify this.

<img width="418" alt="image" src="https://github.com/microsoft/aisdkforsapabap/assets/137146462/835888b2-3cbd-44e4-a1f6-eb029f8c8e3e">
This is the input
<img width="783" alt="SIMPLE issue" src="https://github.com/microsoft/aisdkforsapabap/assets/137146462/7265abe9-5a05-4119-ae0f-1ad4478fc460">

**Comment by gopalnair** [comment]
Hi @Dev4Develop ,
The demo program showcases chat completion API. The Azure Open AI version that supports it is : 2023-03-15-preview

This is documented here : https://microsoft.github.io/aisdkforsapabap/docs/working-with-sdk/chat_completions

Please run the demo program with version string as : 2023-03-15-preview

Let me know how it goes.

Note to self : add text description to parameters :).

**Reply by gopalnair** [reply]
Oh... and one another point... DaVinci model is gpt3. So it won't work with chat completion. 

Rather, please deploy a gpt35turbo or gpt4 model and use that deployment ID.

Best Regards,
Goal Nair.

**Comment by Dev4Develop** [comment]
Im able to see the ouput with gpt-35-turbo and date change. Thanks. 

I dont see gpt4 in the list. Are there major improvements in gpt4 vs gpt-35-turbo.
My requirement is for code completions though.

![image](https://github.com/microsoft/aisdkforsapabap/assets/137146462/bf84e098-c1e5-4f67-8166-711bde892684)

**Comment by gopalnair** [comment]
GPT-4 is certainly more powerful than GPT-3.5-turbo... However, access to GPT-4 is limited by Microsoft, and will require  requesting the same. Once the request is approved, the model will be available in your subscription.
